pageserver/src/tenant/storage_layer/layer_contents/virtual_value.rs

```diff
@@ -0,0 +1,207 @@
+use crate::repository::Value;
+use crate::walrecord::NeonWalRecord;
+use bytes::Bytes;
+use serde::{Deserialize, Serialize};
+use utils::lsn::Lsn;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum VirtualValue {
+    NaturalImage(Bytes),
+    NaturalWalRecord(NeonWalRecord),
+    ClosedLineage {
+        image: Bytes,
+        lsns: Vec<Lsn>,
+        records: Vec<NeonWalRecord>,
+    },
+    ClosedRecLineage {
+        image_rec: NeonWalRecord,
+        lsns: Vec<Lsn>,
+        records: Vec<NeonWalRecord>,
+    },
+    OpenLineage {
+        lsns: Vec<Lsn>,
+        records: Vec<NeonWalRecord>,
+    },
+}
```

Comment:
This would be a great place to document the meaning of these in rustdoc style:

```
/// git commit msg style subject or doxygen summary on the first "paragraph"
///
/// longer text
#[derive(...)]
pub enum VirtualValue {
    /// similarly for each of these
    NaturalImage(Bytes),
    ...
}
```

This should be the priority place to describe these. Based on your PR desc the rest seems to be quite self-describing at least on first glance.

Reply:
Thanks for the added comments, good `NOTE:` as well! Resolving.